DeepCat.js

```diff
@@ -11,7 +11,20 @@
 	var DBname = mw.config.get( 'wgDBname' );
 	var requestUrl = '//tools.wmflabs.org/catgraph-jsonp/' + DBname + '_ns14/traverse-successors%20Category:{0}%20' + maxDepth + '%20' + maxResults;
 
-
+	// hash function for generating hint box cookie token.
+	// see http://erlycoder.com/49/javascript-hash-functions-to-convert-string-into-integer-hash-
+	function djb2Code(str) {
```

Comment:
BTW, if you know some checksumming or hashing function built into jQuery or the MW API, let me know. I checked the docs but didn't find one.